sdk/data/azappconfig/client_test.go

```diff
@@ -1242,3 +1242,243 @@ func TestTagsFilteringEndToEnd(t *testing.T) {
+		value string
+		label string
+		tags  map[string]*string
+	}{
+		{
+			key:   "setting-with-normal-tags",
+			value: "value1",
+			label: "prod",
+			tags: map[string]*string{
+				"env":     to.Ptr("production"),
+				"service": to.Ptr("api"),
+				"version": to.Ptr("1.0"),
+			},
+		},
+		{
+			key:   "setting-with-nil-tag-value",
+			value: "value2",
+			label: "prod",
+			tags: map[string]*string{
+				"env":     to.Ptr("production"),
+				"service": nil, // nil tag value
+				"version": to.Ptr("1.0"),
+			},
+		},
+		{
+			key:   "setting-with-empty-tag-value",
+			value: "value3",
+			label: "prod",
+			tags: map[string]*string{
+				"env":     to.Ptr("production"),
+				"service": to.Ptr(""), // empty string tag value
+				"version": to.Ptr("1.0"),
+			},
+		},
+		{
+			key:   "setting-mixed-tags",
+			value: "value4",
+			label: "staging",
+			tags: map[string]*string{
+				"env":       to.Ptr("staging"),
+				"service":   to.Ptr("web"),
+				"version":   to.Ptr("2.0"),
+				"feature":   nil,                // nil value
+				"debug":     to.Ptr(""),         // empty value
+				"component": to.Ptr("frontend"), // normal value
+			},
+		},
+		{
+			key:   "setting-only-nil-tags",
+			value: "value5",
+			label: "test",
+			tags: map[string]*string{
+				"tag1": nil,
+				"tag2": nil,
+				"tag3": nil,
+			},
+		},
+		{
+			key:   "setting-only-empty-tags",
+			value: "value6",
+			label: "test",
+			tags: map[string]*string{
+				"tag1": to.Ptr(""),
+				"tag2": to.Ptr(""),
+				"tag3": to.Ptr(""),
+			},
+		},
+	}
+
+	// Create all test settings
+	for _, data := range testData {
+		_, err := client.AddSetting(context.Background(), data.key, &data.value, &azappconfig.AddSettingOptions{
+			Label: to.Ptr(data.label),
+			Tags:  data.tags,
+		})
+		require.NoError(t, err)
+	}
+
+	// Test 1: Filter by multiple specific tag values
+	t.Run("MultipleTagsFilter", func(t *testing.T) {
+		selector := azappconfig.SettingSelector{
+			TagsFilter: []string{"env=production", "version=1.0"},
+		}
+
+		var foundKeys []string
+		pager := client.NewListSettingsPager(selector, nil)
+		for pager.More() {
+			page, err := pager.NextPage(context.Background())
+			require.NoError(t, err)
+
+			for _, setting := range page.Settings {
+				if strings.HasPrefix(*setting.Key, "setting-") {
+					foundKeys = append(foundKeys, *setting.Key)
+				}
+			}
+		}
+
+		// Should find settings with both env=production AND version=1.0
+		require.Contains(t, foundKeys, "setting-with-normal-tags")
+		require.Contains(t, foundKeys, "setting-with-nil-tag-value")
+		require.Contains(t, foundKeys, "setting-with-empty-tag-value")
+		require.NotContains(t, foundKeys, "setting-mixed-tags") // has env=staging
+	})
+
+	// Test 2: Filter by tag key regardless of value (including nil/empty)
+	t.Run("TagKeyExistsFilter", func(t *testing.T) {
+		// Note: This test depends on how the backend handles tag key existence
+		// Some implementations might not return nil-valued tags
```

Comment:
Backend does not support filtering by tag key only (or tag existence). The only supported scenario is filtering by exact match (case sensitive) of tag name and tag value both.